tests/rptest/tests/quota_management_test.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.clients.kafka_cli_tools import KafkaCliTools, KafkaCliToolsError
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.util import expect_exception
+
+
+def expect_kafka_cli_error_msg(error_msg: str):
+    return expect_exception(KafkaCliToolsError,
+                            lambda e: error_msg in e.output)
+
+
+class QuotaManagementTest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.kafka_cli = KafkaCliTools(self.redpanda)
+
+    @cluster(num_nodes=1)
+    def test_not_supported(self):
+        with expect_kafka_cli_error_msg(
+                "Unsupported version - not yet implemented"):
+            self.kafka_cli.alter_quota_config(
+                "--entity-type clients --entity-name client_1",
+                to_add={"consumer_byte_rate": 10240})
```

Comment:
question: what would this do without the new API stubs?

Reply:
Another exception is thrown then:
```
java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support DESCRIBE_CLIENT_QUOTAS
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
        at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:180)
        at kafka.admin.ConfigCommand$.getAllClientQuotasConfigs(ConfigCommand.scala:649)
        at kafka.admin.ConfigCommand$.getClientQuotasConfig(ConfigCommand.scala:630)
        at kafka.admin.ConfigCommand$.alterQuotaConfigs(ConfigCommand.scala:470)
        at kafka.admin.ConfigCommand$.alterConfig(ConfigCommand.scala:422)
        at kafka.admin.ConfigCommand$.processCommand(ConfigCommand.scala:327)
        at kafka.admin.ConfigCommand$.main(ConfigCommand.scala:98)
        at kafka.admin.ConfigCommand.main(ConfigCommand.scala)
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support DESCRIBE_CLIENT_QUOTAS
```